GiniVision/Classes/Core/Screens/Camera/CameraPreviewViewController.swift

```diff
@@ -0,0 +1,292 @@
+        return UIImageNamedPreferred(named: "cameraFocusSmall")
+    }
+    
+    fileprivate var cameraFocusLarge: UIImage? {
+        return UIImageNamedPreferred(named: "cameraFocusLarge")
+    }
+    
+    fileprivate var defaultImage: UIImage? {
+        return UIImageNamedPreferred(named: "cameraDefaultDocumentImage")
+    }
+    
+    fileprivate enum CameraState {
+        case valid, notValid
+    }
+
+    lazy var previewView: CameraPreviewView = {
+        let previewView = CameraPreviewView()
+        previewView.translatesAutoresizingMaskIntoConstraints = false
+        (previewView.layer as? AVCaptureVideoPreviewLayer)?.videoGravity = .resize
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(focusAndExposeTap))
+        previewView.addGestureRecognizer(tapGesture)
+        return previewView
+    }()
+    
+    init(giniConfiguration: GiniConfiguration = .shared) {
+        self.giniConfiguration = giniConfiguration
+        super.init(nibName: nil, bundle: nil)
+        
+        setupCamera(giniConfiguration: giniConfiguration)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    override func loadView() {
+        super.loadView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        
+        if let validCamera = camera {
+            cameraState = .valid
+            previewView.session = validCamera.session
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(subjectAreaDidChange),
+                                                   name: NSNotification.Name.AVCaptureDeviceSubjectAreaDidChange,
+                                                   object: camera?.videoDeviceInput?.device)
+        }
+        previewView.drawGuides(withColor: giniConfiguration.cameraPreviewCornerGuidesColor)
+        
+        view.insertSubview(previewView, at: 0)
+        Constraints.pin(view: previewView, toSuperView: view)
+
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        updatePreviewViewOrientation() // Video orientation should be updated once the view has been loaded
+    }
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        camera?.start()
+    }
+    
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        camera?.stop()
+    }
+    
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            guard let `self` = self else {
```

Comment:
Is this guard necessary? Why can't you just have `self?.updatePreviewViewOrientation()`?

Reply:
Without this it doesn't look very well, since the animation doesn't happen at the same time